### PR TITLE
Make code more explicit and don't print duplicate errors in monitoring_read

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1524/)

https://ci.cfengine.com/job/pr-pipeline/1524/

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1525/)

https://ci.cfengine.com/job/pr-pipeline/1525/

**Comment by olehermanse** [comment]
@vpodzime please review. (need to expand context to see the whole function).

File: libpromises/monitoring_read.c [hunk 1]
```diff
@@ -164,7 +164,19 @@ void Nova_LoadSlots(void)
...
-            fscanf(f, "%*[^\n]\n");
+            int r;
+            do
+            {
+                r = fgetc(f);
+                if (r == EOF)
+                {
+                    break;
+                }
+            } while (r != (int)'\n');
```
Comment (vpodzime) [inline libpromises/monitoring_read.c]: Please move change the `if-break` into a second condition in the `do-while`.

**Comment by olehermanse** [comment]
@cf-bottom jenkins

File: libpromises/monitoring_read.c (hunk 1 above)
Comment (olehermanse) [reply libpromises/monitoring_read.c]: Thanks, fixed.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1532/)

https://ci.cfengine.com/job/pr-pipeline/1532/

**Pull request merged**